### ISSUE TITLE
Chrome driver does not close Driver Service Executor and Driver Command Executor threads 

### ISSUE THREAD

**Issue by ajurge** [SeleniumHQ/selenium#9789]
## 🐛 Bug Report

Chrome driver does not close Driver Service Executor and Driver Command Executor threads ,and after some time produces  
```
java.lang.OutOfMemoryError: unable to create native thread: possibly out of memory or process/resource limits reached
```

####

Error on Linux
```
Build info: version: '4.0.0-beta-4', revision: '29f46d02dd'
   System info: host: '', ip: '', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-142-generic', java.version: '11.0.12'
   Driver info: org.openqa.selenium.chrome.ChromeDriver
   Command: [null, newSession {capabilities=[Capabilities {browserName: chrome, goog:chromeOptions: {args: [--headless, window-size=1400,1400, user-agent=test ..., --no-sandbox, --disable-dev-shm-usage], extensions: []}}], desiredCapabilities=Capabilities {--whitelisted-ips: , browserName: chrome, ensureCleanSession: true, goog:chromeOptions: {args: [--headless, window-size=1400,1400, user-agent=test ..., --no-sandbox, --disable-dev-shm-usage], extensions: []}}}]
       at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:126)
       at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:84)
       at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:62)
       at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:156)
       at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:162)
       at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:137)
       at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:612)
       at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:244)
       at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:165)
       at org.openqa.selenium.chromium.ChromiumDriver.<init>(ChromiumDriver.java:89)
       at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:99)
       at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:86)
       at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:75)
       ...
       at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
       at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779)
       at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
       at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
       at net.bull.javamelody.MonitoringSpringInterceptor.invoke(MonitoringSpringInterceptor.java:76)
       at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
       at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)
       at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)
       ...
       at jdk.internal.reflect.GeneratedMethodAccessor100.invoke(Unknown Source)
       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
       at java.base/java.lang.reflect.Method.invoke(Unknown Source)
       at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84)
       at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
       at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
       at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
       at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)

```

JConsole Threads (>200 Live Threads after 7 min, scheduled service execution fixed period  10s):
 
On Windows
```
Name: Driver Service Executor
State: WAITING on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@41b0f2fc
Total blocked: 0  Total waited: 29

Stack trace: 
java.base@11.0.2/jdk.internal.misc.Unsafe.park(Native Method)
java.base@11.0.2/java.util.concurrent.locks.LockSupport.park(LockSupport.java:194)
java.base@11.0.2/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2081)
java.base@11.0.2/java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:433)
java.base@11.0.2/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1054)
java.base@11.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1114)
java.base@11.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
java.base@11.0.2/java.lang.Thread.run(Thread.java:834)




Name: Driver Command Executor
State: WAITING on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@4909a6ea
Total blocked: 0  Total waited: 3

Stack trace: 
java.base@11.0.2/jdk.internal.misc.Unsafe.park(Native Method)
java.base@11.0.2/java.util.concurrent.locks.LockSupport.park(LockSupport.java:194)
java.base@11.0.2/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2081)
java.base@11.0.2/java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:433)
java.base@11.0.2/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1054)
java.base@11.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1114)
java.base@11.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
java.base@11.0.2/java.lang.Thread.run(Thread.java:834)
```

Similar issues:

1. https://github.com/SeleniumHQ/selenium/issues/9224 
2. https://github.com/SeleniumHQ/selenium/issues/9153 
2. https://github.com/SeleniumHQ/selenium/issues/8028
3. https://github.com/SeleniumHQ/selenium/issues/7989

## To Reproduce

This is just a possible way to reproduce it (we have a very similar setup in our scheduled service), i have not tested it: 

`seleniumVersion = '4.0.0-beta-4'`

```      
for (int i = 0; i < 6000; i++) {
   var driverContext = new EventFiringWebDriver(new ChromeDriver());
   driverContext.close();
   driverContext.quit();
}
```

## Expected behavior
Driver Service Executor and Driver Command Executor threads are closed.

## Environment

OS: Linux, Windows 10 
Browser: Chrome 
Browser version: 93.0.4577.63
Browser Driver version: ChromeDriver 93.0.4577.15
Language Bindings version: Java 11.0.10
Selenium Grid version (if applicable): 4.0.0-beta-4  
Netty version (if applicable): 4.1.67.Final

**Comment by ajurge** [comment]
This seems to be solved in Selenium version 4.0.0-rc-1, so i guess this bug can be closed.

**Comment by diemol** [comment]
Exactly, this was fixed in rc-1.
